src/shared/Core/Interop/Linux/LinuxSessionManager.cs

```diff
@@ -0,0 +1,64 @@
+using GitCredentialManager.Interop.Posix;
+
+namespace GitCredentialManager.Interop.Linux;
+
+public class LinuxSessionManager : PosixSessionManager
+{
+    private bool? _isWebBrowserAvailable;
+
+    public LinuxSessionManager(IEnvironment env, IFileSystem fs) : base(env, fs)
+    {
+        PlatformUtils.EnsureLinux();
+    }
+    
+    public override bool IsWebBrowserAvailable
+    {
+        get
+        {
+            return _isWebBrowserAvailable ??= GetWebBrowserAvailable();
+        }
+    }
+    
+    private bool GetWebBrowserAvailable()
+    {
+        // If this is a Windows Subsystem for Linux distribution we may
+        // be able to launch the web browser of the host Windows OS.
+        if (WslUtils.IsWslDistribution(Environment, FileSystem, out _))
+        {
+            // We need a shell execute handler to be able to launch to browser
+            if (!BrowserUtils.TryGetLinuxShellExecuteHandler(Environment, out _))
+            {
+                return false;
+            }
+
+            //
```

Comment:
Out of curiosity, when should one add `// <empty line>` before a comment versus `// <comment>`?

Reply:
It's more personal taste. When a comment block ends up feeling like a novel with multiple paragraphs, I think it's easier to read (and stands out more as something important to understand!) with surrounding whitespace.

So let's say "a comment block with multiple paragraphs should have a `//` blank line at start and end"?